KsApi/Service.swift

```diff
@@ -351,6 +363,30 @@ public struct Service: ServiceType {
       .flatMap(UserEnvelope<GraphUserEmail>.envelopeProducer(from:))
   }
 
+  public func fetchGraphUserEmailCombine()
+    -> AnyPublisher<UserEnvelope<GraphUserEmail>, ErrorEnvelope> {
+    GraphQL.shared.client
+      .fetch(query: GraphAPI.FetchUserEmailQuery())
+      // TODO: make this a custom extension, we'll want to reuse this pattern
+      .tryMap { (data: GraphAPI.FetchUserEmailQuery.Data) -> UserEnvelope<GraphUserEmail> in
+        guard let envelope = UserEnvelope<GraphUserEmail>.userEnvelope(from: data) else {
+          throw ErrorEnvelope.couldNotParseJSON
+        }
+
+        return envelope
+      }
+      .mapError { rawError in
+
+        if let error = rawError as? ErrorEnvelope {
+          return error
+        }
+
+        return ErrorEnvelope.couldNotParseJSON
```

Comment:
Not clear if there would be a better error to use here. This would be the case when something in userEnvelope(from: data) threw an exception.

Reply:
Good question. We could probably use a more verbose error. It looks like the rest of the app currently returns this `couldNotParseJSON` enveloper